signer/src/transaction_coordinator.rs

```diff
@@ -1672,76 +1695,295 @@ where
+    /// time as well. We need to do this because deposit requests are locked
+    /// using OP_CSV, which lock up coins based on block height or
+    /// multiples of 512 seconds measure by the median time past.
+    #[tracing::instrument(skip_all)]
+    pub async fn get_eligible_pending_deposit_requests<DB>(
+        storage: &DB,
+        context_window: u16,
+        params: &GetPendingRequestsParams<'_>,
+    ) -> Result<Vec<utxo::DepositRequest>, Error>
+    where
+        DB: DbRead,
+    {
+        tracing::debug!("fetching eligible deposit requests");
+        let mut eligible_deposits: Vec<utxo::DepositRequest> = Vec::new();
+
+        // First, we fetch pending deposit requests with initial filtering
+        // done by the storage layer.
+        let pending_deposit_requests = storage
+            .get_pending_accepted_deposit_requests(
+                params.bitcoin_chain_tip.as_ref(),
+                context_window,
+                params.signature_threshold,
+            )
+            .await?;
+
+        // If there are no pending deposit requests, we can exit early.
+        if pending_deposit_requests.is_empty() {
+            tracing::debug!("no pending deposit requests eligible for consideration found");
+            return Ok(eligible_deposits);
+        }
+
+        // Iterate through each deposit request, fetch its _actual_ votes from
+        // storage and
+        for req in pending_deposit_requests {
             let votes = storage
-                .get_withdrawal_request_signer_votes(&req.qualified_id(), aggregate_key)
+                .get_deposit_request_signer_votes(&req.txid, req.output_index, params.aggregate_key)
                 .await?;
 
-            let withdrawal = utxo::WithdrawalRequest::from_model(req, votes);
-            withdrawals.push(withdrawal);
+            let deposit = utxo::DepositRequest::from_model(req, votes);
+            eligible_deposits.push(deposit);
         }
 
+        Ok(eligible_deposits)
+    }
+
+    /// Fetches pending deposit and withdrawal requests from storage and filters
+    /// them based on consensus rules defined in #741 and [**missing**: deposit
+    /// consensus ticket?].
+    #[tracing::instrument(skip_all)]
+    pub async fn get_pending_requests(
+        &self,
+        bitcoin_chain_tip: &model::BitcoinBlockRef,
+        stacks_chain_tip: &model::StacksBlockHash,
+        aggregate_key: &PublicKey,
+        signer_public_keys: &BTreeSet<PublicKey>,
+    ) -> Result<Option<utxo::SbtcRequests>, Error> {
+        tracing::info!("preparing pending requests for processing");
+
+        let storage = self.context.get_storage();
+        let config = self.context.config();
+
+        // Get the current sBTC limits (caps).
+        let sbtc_limits = self.context.state().get_current_limits();
+
+        // Fetch pending deposit and withdrawal requests from storage.
+        let (deposits, withdrawals) = {
```

Comment:
The body seems short enough to get rid of this and just compute the values, is there a reason to prefer the added inner scope here?

Reply:
It was due to the borrow of `sbtc_limits` which is then moved when creating the `SbtcRequests` below, so the immutable borrow needs to drop first.

But thinking after, I could make that owned and then move it from `params` to `SbtcRequests` when done.

Reply:
It was due to some other borrowed field I had and removed after-the-fact I guess, because it worked fine now :) Changed in [55214dd](https://github.com/stacks-network/sbtc/pull/1399/commits/55214dd1d869dfc1bf2158d41b643be04c2a30bb)